File: app/src/main/java/com/commit451/lifeline/sample/CheckIfForegroundService.java
package com.commit451.lifeline.sample;

import android.app.IntentService;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;
import android.widget.Toast;

import com.commit451.lifeline.Lifeline;

import java.util.concurrent.TimeUnit;

/**
 * Check if the app is in the foreground
 */
public class CheckIfForegroundService extends IntentService {

    public static void check(Context context) {
        Intent intent = new Intent(context, CheckIfForegroundService.class);
        context.startService(intent);
    }

    public CheckIfForegroundService() {
        super(CheckIfForegroundService.class.getSimpleName());
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        postOnMainThread(new Runnable() {
            @Override
            public void run() {
                Toast.makeText(CheckIfForegroundService.this, "In foreground: " + Lifeline.isInForeground(), Toast.LENGTH_SHORT).show();
                Log.d("TEST", "Current visible activity: " + Lifeline.getCurrentVisibleActivity());
                Log.d("TEST", "Current created activity: " + Lifeline.getCurrentCreatedActivity());
            }
        });
        try {
            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
            check(getApplicationContext());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void postOnMainThread(Runnable runnable) {
        Handler handler = new Handler(Looper.getMainLooper());
        handler.post(runnable);
    }
}

File: app/src/main/java/com/commit451/lifeline/sample/CheckIfForegroundService.kt
package com.commit451.lifeline.sample

import android.app.IntentService
import android.content.Context
import android.content.Intent
import android.os.Handler
import android.os.Looper
import android.util.Log
import android.widget.Toast

import com.commit451.lifeline.Lifeline

import java.util.concurrent.TimeUnit

/**
 * Check if the app is in the foreground
 */
class CheckIfForegroundService : IntentService(CheckIfForegroundService::class.java.simpleName) {

    override fun onHandleIntent(intent: Intent?) {
        postOnMainThread(Runnable {
            Toast.makeText(this@CheckIfForegroundService, "In foreground: " + Lifeline.isInForeground(), Toast.LENGTH_SHORT).show()
            Log.d("TEST", "Current visible activity: " + Lifeline.currentVisibleActivity())
            Log.d("TEST", "Current created activity: " + Lifeline.currentCreatedActivity())
        })
        try {
            Thread.sleep(TimeUnit.SECONDS.toMillis(10))
            check(applicationContext)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    private fun postOnMainThread(runnable: Runnable) {
        val handler = Handler(Looper.getMainLooper())
        handler.post(runnable)
    }

    companion object {

        fun check(context: Context) {
            val intent = Intent(context, CheckIfForegroundService::class.java)
            context.startService(intent)
        }
    }
}
